src/routes/webhooks/paddle.ts

```diff
@@ -74,6 +82,39 @@ const updateUserSubscription = async ({
     );
     return false;
   }
+
+  const { gifter_id: gifterId, duration } = customData;
+  const durationTime = duration && parseInt(duration);
+  const isGift = !!durationTime && gifterId;
+  if (isGift) {
+    if (userId === gifterId) {
+      logger.error({ type: 'paddle', data }, 'User and gifter are the same');
+      return false;
+    }
+
+    if (!durationTime || durationTime <= 0) {
+      logger.error({ type: 'paddle', data }, 'Invalid duration');
+      return false;
+    }
+
+    const gifterUser = await con
+      .getRepository(User)
+      .findOneBy({ id: gifterId });
+    if (!gifterUser) {
+      logger.error({ type: 'paddle', data }, 'Gifter user not found');
+      return false;
+    }
```

Comment:
Might be good to add alerts for some of these (CC @capJavert )

Reply:
yeah agree, we can do it once this is deployed 👌

Reply:
@ilasw can you make note to setup this alerts after the feature goes live?

Reply:
Yes ✔️ 
Where do we setup those alerts? 👀

Reply:
https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/1234567185/Alerting+and+incident+management